modules/function/src/main/java/com/opengamma/strata/function/marketdata/MarketDataRatesProvider.java

```diff
@@ -99,15 +102,18 @@ public double fxRate(Currency baseCurrency, Currency counterCurrency) {
   //-------------------------------------------------------------------------
   @Override
   public DiscountFactors discountFactors(Currency currency) {
-    return marketData.getValue(DiscountFactorsKey.of(currency));
+    Curve curve = marketData.getValue(DiscountCurveKey.of(currency));
```

Comment:
`MarketData.getValue` never returns null. It throws `IllegalArgumentException` for an unknown key.